logger/logObject.go

```diff
@@ -0,0 +1,186 @@
+package logger
+
+import (
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type LogObject interface {
+	// LogName returns the name of the object to log
+	LogName() string
+	// LogProperties returns the fields to log
+	// NOTE: Avoid using struct as a value as it will trigger a reflection field
+	// only use if it's mandatory
+	LogProperties() map[string]any
+}
+
+// MapObject create LogObject with data
+func MapObject(logName string, data map[string]any) LogObject {
+	return &wrapper{
+		Name:   logName,
+		Fields: data,
+	}
+}
+
+// wrapper wraps any object as a loggable object
+type wrapper struct {
+	Name   string
+	Fields map[string]any
+}
+
+// LogName returns the name of the object to log
+func (w *wrapper) LogName() string {
+	return w.Name
+}
+
+// LogProperties returns the fields to log
+func (w *wrapper) LogProperties() map[string]any {
+	return w.Fields
+}
+
+// ErrObject returns an LogObject containing error information
+func ErrObject(err error) LogObject {
+	fields := map[string]any{
+		"error": "empty error",
+	}
+	if err != nil {
+		fields["error"] = err.Error()
+	}
+
+	return &wrapper{
+		Name:   "errorMsg",
+		Fields: fields}
+}
+
+// InfoObject returns an LogObject containing information
+func InfoObject(msg string) LogObject {
+	return &wrapper{
+		Name: "infoMsg",
+		Fields: map[string]any{
+			"msg": msg,
+		},
+	}
+}
+
+// InfoObject returns an LogObject containing information
+func WarnObject(err error) LogObject {
+	fields := map[string]any{
+		"error": "empty error",
```

Comment:
Can we define this "empty error" string as a const?

As well, I think it would be good to allow setting the fields key dynamically as we may not always want an error message here.

Reply:
mm I don't think this function wants that setting. WarnObject should only be used to log an error as a warning, if we would want to log a message instead (not an error) we would use `InfoObject` or define a custom object using `mapObject`.

For the const 100%

Reply:
OK that's reasonable.